### PR TITLE
Feature get flashcards

### PR DESCRIPTION
**Pull request by Duskt**
- Add GET for /flashcard/get to request flashcards from folder id (api)
- Add function to request flashcards (web)
- Review page loads flashcards from selected folders (web)

### CODE REVIEW

File: web/pages/review.tsx [hunk 1]
```diff
@@ -6,38 +6,39 @@ import { getFlashcards } from "@src/getFlashcards";
...
+        ).toString().length;
+        return Math.max(6, fidLength * 2 + 2);
+    }
+
+    const onSelectedFoldersChange = useCallback(
+        async (folderIds: number[]) => {
+            let flashcards: Flashcard[] = [];
+            if (!auth.user?.id_token) return flashcards;
+            for (const fId of folderIds) {
+                flashcards = flashcards.concat(
```
Comment (jacob-horton) [inline web/pages/review.tsx:32]: Could use spread here? Not sure what's better?

File: web/src/components/FolderListView.tsx [hunk 2]
```diff
@@ -42,6 +42,7 @@ const FolderListView: React.FC<FolderListViewProps> = ({
     }, [auth.user, folders]);
 
     useEffect(() => {
+        // must be useCallback, otherwise it will rerender constantly. enforce?
```
Comment (jacob-horton) [inline web/src/components/FolderListView.tsx:45]: I like to add `NOTE: ` to things like this to make it clear its extra information, not just explaining the code

File: web/pages/review.tsx (hunk 1 above)
Comment (jacob-horton) [reply web/pages/review.tsx:32]: Turns out concat is more performant as it does not iterate over all the elements, and has array specific optimisations

File: web/src/components/FolderListView.tsx (hunk 2 above)
Comment (Duskt) [reply web/src/components/FolderListView.tsx:45]: Will follow this convention :thumbsup:

**Review by jacob-horton — APPROVED** [review]

**Pull request merged**